### PR TITLE
[Checkbox][Select][TextField] error border updates

### PR DESCRIPTION
**Pull request by sophschneider**
### WHY are these changes introduced?

Resolves [#128](https://github.com/Shopify/polaris-summer-editions/issues/128)

### WHAT is this pull request doing?

* Updates error border token to `--p-color-border-critical-strong-experimental`
* `TextField` and `Select` use `--p-border-width-1-experimental` which is technically `0.66px` but rounds to `0.5px` in the browser

|Component|gif|
|-|-|
|Checkbox|Screen Recording 2023-06-09 at 2 53 15 PM|
|Select| Screen Recording 2023-06-09 at 3 01 34 PM|
|TextField|Screen Recording 2023-06-09 at 2 51 51 PM|

### How to 🎩

[Checkbox](https://5d559397bae39100201eedc1-veswztuttv.chromatic.com/?path=/story/all-components-checkbox--error&globals=polarisSummerEditions2023:true)
[Select](https://5d559397bae39100201eedc1-veswztuttv.chromatic.com/?path=/story/all-components-select--with-separate-validation-error&globals=polarisSummerEditions2023:true)
[TextField](https://5d559397bae39100201eedc1-veswztuttv.chromatic.com/?path=/story/all-components-textfield--with-separate-validation-error&globals=polarisSummerEditions2023:true)

### 🎩 checklist

- [x] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [x] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)

### CODE REVIEW

File: polaris-react/src/components/RadioButton/RadioButton.scss [hunk 1]
```diff
@@ -18,7 +18,7 @@
       @include no-focus-ring;
       outline: var(--p-border-width-2) solid
         var(--p-color-border-interactive-focus);
-      outline-offset: var(--p-space-05);
+      outline-offset: var(--p-space-025);
```
Comment (sophschneider) [inline polaris-react/src/components/RadioButton/RadioButton.scss:21]: Note `RadioButton` doesn't have an error state. Corrected this offset for focus ring to have a space of 1px to match figma

|Before|After|
|-|-|
|Screenshot 2023-06-09 at 3 18 28 PM|Screenshot 2023-06-09 at 3 18 37 PM|

File: polaris-react/src/components/Checkbox/Checkbox.scss [hunk 2]
```diff
@@ -18,7 +18,7 @@
       #{$se23} & {
         outline: var(--p-border-width-2) solid
           var(--p-color-border-interactive-focus);
-        outline-offset: var(--p-space-05);
+        outline-offset: var(--p-space-025);
```
Comment (sophschneider) [inline polaris-react/src/components/Checkbox/Checkbox.scss:21]: |Before|After|
|-|-|
|Screenshot 2023-06-09 at 3 17 00 PM|Screenshot 2023-06-09 at 3 17 21 PM|

**Comment by sophschneider** [comment]
FYI I noticed our experimental token value for `--p-border-width-1-experimental` of `0.66px` gets rounded to `0.5px` in the browser. Should we change the token value to `0.5px` to avoid confusion or keep it as is?

File: polaris-react/src/components/Checkbox/Checkbox.scss (hunk 2 above)
Comment (kyledurand) [reply polaris-react/src/components/Checkbox/Checkbox.scss:21]: I changed this to 025 yesterday but @gwyneplaine changed it back overnight. Was causing a legit issue https://github.com/Shopify/polaris/pull/9426#event-9485668849

Comment (sophschneider) [reply polaris-react/src/components/Checkbox/Checkbox.scss:21]: the size issue? its still working with the `025` on this PR! supposed to be `16px`x`16px` on `md-up` and `18px`x`18px` on small screens

**Review by kyledurand — APPROVED** [review]

**Review by bernardojoaogarcia — APPROVED** [review]

**Review by laurkim — APPROVED** [review]
Nice 💯

**Review by sarahill — APPROVED** [review]

**Review by aaronccasanova — APPROVED** [review]

**Pull request merged**